第四次实验/task1.sh

```diff
@@ -0,0 +1,130 @@
+#!bin/bash
```

Comment:
In task1.sh line 7:
    echo "Usage: bash "$0" [options]"
                       ^-- SC2027: The surrounding quotes actually unquote this. Remove or escape them.
                       ^-- SC2086: Double quote to prevent globbing and word splitting.


In task1.sh line 22:
    imgs=($(find "$1" ! -path ""$1"output/*" -name "*.jpeg"))
                                ^-- SC2027: The surrounding quotes actually unquote this. Remove or escape them.
                                ^-- SC2086: Double quote to prevent globbing and word splitting.


In task1.sh line 24:
        $(convert "$img" -quality "$2" ""$1"output/compressq_${img:${#1}}")
        ^-- SC2091: Remove surrounding $() to avoid executing output.
                                         ^-- SC2027: The surrounding quotes actually unquote this. Remove or escape them.
                                         ^-- SC2086: Double quote to prevent globbing and word splitting.


In task1.sh line 30:
    imgs=($(find "$1" ! -path ""$1"output/*" -name "*.jpeg" -o -name "*.png" -o -name "*.svg"))
                                ^-- SC2027: The surrounding quotes actually unquote this. Remove or escape them.
                                ^-- SC2086: Double quote to prevent globbing and word splitting.


In task1.sh line 32:
        $(convert "$img" -resize "$2" ""$1"output/compressr_${img:${#1}}")
        ^-- SC2091: Remove surrounding $() to avoid executing output.
                                        ^-- SC2027: The surrounding quotes actually unquote this. Remove or escape them.
                                        ^-- SC2086: Double quote to prevent globbing and word splitting.


In task1.sh line 38:
    imgs=($(find "$1" ! -path ""$1"output/*" -name "*.png"))
                                ^-- SC2027: The surrounding quotes actually unquote this. Remove or escape them.
                                ^-- SC2086: Double quote to prevent globbing and word splitting.


In task1.sh line 41:
        $(convert -background '#0008' -fill white -gravity center -size ${width}x30 caption:"$2" "$img" +swap -gravity south -composite ""$1"output/addwm_${img:${#1}}")
        ^-- SC2091: Remove surrounding $() to avoid executing output.
                                                                        ^-- SC2086: Double quote to prevent globbing and word splitting.
                                                                                                                                          ^-- SC2027: The surrounding quotes actually unquote this. Remove or escape them.
                                                                                                                                          ^-- SC2086: Double quote to prevent globbing and word splitting.


In task1.sh line 51:
        $(cp "$img" "$newname")
        ^-- SC2091: Remove surrounding $() to avoid executing output.


In task1.sh line 59:
        fullname=$(basename $img)
                            ^-- SC2086: Double quote to prevent globbing and word splitting.


In task1.sh line 60:
        filename=$(echo $fullname | cut -d . -f1)
                        ^-- SC2086: Double quote to prevent globbing and word splitting.


In task1.sh line 61:
        extension=$(echo $fullname | cut -d . -f2)
                         ^-- SC2086: Double quote to prevent globbing and word splitting.


In task1.sh line 63:
        $(cp "$img" "$newname")
        ^-- SC2091: Remove surrounding $() to avoid executing output.


In task1.sh line 68:
    imgs=($(find "$1" ! -path ""$1"output/*" -name "*.png" -o -name "*.svg"))
                                ^-- SC2027: The surrounding quotes actually unquote this. Remove or escape them.
                                ^-- SC2086: Double quote to prevent globbing and word splitting.


In task1.sh line 70:
        fullname=$(basename $img)
                            ^-- SC2086: Double quote to prevent globbing and word splitting.


In task1.sh line 71:
        filename=$(echo $fullname | cut -d . -f1)
                        ^-- SC2086: Double quote to prevent globbing and word splitting.


In task1.sh line 72:
        extension=$(echo $fullname | cut -d . -f2)
                         ^-- SC2086: Double quote to prevent globbing and word splitting.


In task1.sh line 73:
        newname=${1}"output/"${filename}".svg"
                             ^-- SC2027: The surrounding quotes actually unquote this. Remove or escape them.


In task1.sh line 74:
        $(convert "$img" "$newname")
        ^-- SC2091: Remove surrounding $() to avoid executing output.